integration-libs/punchout/root/services/punchout-detection.service.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * SPDX-FileCopyrightText: 2025 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Location } from '@angular/common';
+import { Injectable, inject } from '@angular/core';
+import { PUNCHOUT_SESSION_PAGE_URL } from '../model';
+
+@Injectable({ providedIn: 'root' })
+export class PunchoutDetectionService {
+  protected location = inject(Location);
+
+  /**
+   * Check if browser url is the punchout initial session page.
+   * With default config, the expected url shape is '/punchout/cxml/session?abcd'.
+   * @returns boolean
+   */
+  isPunchoutSessionPage(): boolean {
+    const urlSections = this.location.path().split('?');
+    return (
+      urlSections.length > 1 &&
```

Comment:
Should we check length first?

Reply:
Alternative:
```
/**
 * Checks if the browser URL is the punchout session page.
 * Default URL shape: '/punchout/cxml/session?abcd'.
 * @returns boolean
 */
isPunchoutSessionPage(): boolean {
  const [path] = this.location.path().split('?');
  return path.includes(PUNCHOUT_SESSION_PAGE_URL);
}
```

Reply:
length is checked first, which avoid to run the 'includes' method.
note that location.path never returns null/undefined so I didn't add any check for it.